File: algorithm/intersection_of_two_arrays/intersection_of_two_arrays.go
package main

import (
	"fmt"
	"sort"
)

func intersection(nums1 []int, nums2 []int) []int {
	sort.IntSlice(nums1).Sort()
	sort.IntSlice(nums2).Sort()
	fmt.Println(nums1)
	fmt.Println(nums2)
	numMap := make(map[int]int)
	for i, j := 0, 0; i < len(nums1) && j < len(nums2); {
		if nums1[i] == nums2[j] {
			numMap[nums1[i]] += 1
			j++
		} else if nums1[i] > nums2[j] {
			j++
			continue
		}
		i++
	}
	res := make([]int, 0, len(numMap))
	for k := range numMap {
		res = append(res, k)
	}
	return res
}

func main() {
	fmt.Println(intersection([]int{9, 1, 2, 2, 3, 1}, []int{2, 2, 8, 7, 13, 3, 9}))
}

File: algorithm/intersection_of_two_arrays/intersection_of_two_arrays.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        return list(set(nums1) & set(nums2))


if __name__ == '__main__':
    s = Solution()
    print s.intersection(range(10), [x for x in range(10) if x % 2 == 0])
